From 28828c2ab4571754e8eda71773d451eb7b19c61f Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:38:43 -0400
Subject: [alsa] pcm: simplify snd_pcm_tstamp()

Message-id: <140972823573-386-patchset-jkysela@redhat.com>
Patchwork-id: 91082
O-Subject: [RHEL 7.1 PATCH 385/442] ALSA: pcm: simplify snd_pcm_tstamp()
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

No functional change.

Acked-by: Jaroslav Kysela <perex@perex.cz>
Reviewed-by: Mark Brown <broonie@linaro.org>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 2408c219df9e7a47680d43ed7877aaaec3feca1b)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/core/pcm_native.c b/sound/core/pcm_native.c
index ade74c4..d1afafe 100644
--- a/sound/core/pcm_native.c
+++ b/sound/core/pcm_native.c
@@ -2540,9 +2540,7 @@ static int snd_pcm_tstamp(struct snd_pcm_substream *substream, int __user *_arg)
   return -EFAULT;
  if (arg < 0 || arg > SNDRV_PCM_TSTAMP_TYPE_LAST)
   return -EINVAL;
- runtime->tstamp_type = SNDRV_PCM_TSTAMP_TYPE_GETTIMEOFDAY;
- if (arg == SNDRV_PCM_TSTAMP_TYPE_MONOTONIC)
-  runtime->tstamp_type = SNDRV_PCM_TSTAMP_TYPE_MONOTONIC;
+ runtime->tstamp_type = arg;
  return 0;
 }
   
-- 
1.7.1